### PR TITLE
Avoid rendering the render target texture.

### PR DESCRIPTION
**Pull request by keianhzo**
Chromium throws an error when trying to render the render target texture.
Fixes https://github.com/mozilla/hubs/issues/4438

### CODE REVIEW

**Review by netpro2k — COMMENTED** [review]
Nice catch. I think this solution should work and should be pretty fast, but I wonder if we should just put render target on their own layer instead, Then we just make sure the video-texture-source cameras do not render that layer.

**Comment by keianhzo** [comment]
I wonder if we wouldn't be over complicating the solution. All the scene cameras would need to render the default layer and the specific target object layer to render the whole scene, then the the render target would need to render only the default layer. So we would need to create all scene objects with both layers enabled just for the default cameras to render the scene and then make the render target just to render the default one.

Alternatively we could just disable all target object layers in the object hierarchy before rendering and then enabling them again which is pretty much the same as toggling their visibility.

**Comment by keianhzo** [comment]
Updated to using layers.

File: src/components/video-texture-target.js [hunk 1]
```diff
@@ -106,6 +107,12 @@ AFRAME.registerComponent("video-texture-target", {
   },
 
   init() {
+    // Make video-texture-target objects inivisible before rendering to the frame buffer
+    // Chromium checks for loops when drawing to a framebuffer so if we don't exclude the objects
+    // that are using that rendertarget's texture we get an error. Firefox does not check.
+    // https://chromium.googlesource.com/chromium/src/+/460cac969e2e9ac38a2611be1a32db0361d88bfb/gpu/command_buffer/service/gles2_cmd_decoder.cc#9516
+    this.el.object3D.traverse(o => o.layers.set(Layers.CAMERA_LAYER_VIDEO_TEXTURE_TARGET));
```
Comment (netpro2k) [inline src/components/video-texture-target.js:114]: Probably doesn't really effect anything but this should probably be removed when the component is removed.

File: src/systems/camera-system.js [hunk 2]
```diff
@@ -207,6 +204,9 @@ export class CameraSystem {
     this.mode = CAMERA_MODE_SCENE_PREVIEW;
     this.snapshot = { audioTransform: new THREE.Matrix4(), matrixWorld: new THREE.Matrix4() };
     this.audioSourceTargetTransform = new THREE.Matrix4();
+    scene.addEventListener("cameraready", ({ detail: { cameraEl } }) => {
+      cameraEl.getObject3D("camera").layers.mask |= LayerMasks.CAMERA_LAYER_VIDEO_TEXTURE_TARGET_MASK;
```
Comment (netpro2k) [inline src/systems/camera-system.js:208]: Can't you use `.enable(Layers.CAMERA_LAYER_VIDEO_TEXTURE_TARGET)` here?

File: src/components/layers.js [hunk 3]
```diff
@@ -1,6 +1,17 @@
 export const Layers = {
   // Layers 0 - 2 reserverd by ThreeJS and AFrame.
-  reflection: 3
+  CAMERA_LAYER_REFLECTION: 3,
+  CAMERA_LAYER_INSPECT: 4,
+  CAMERA_LAYER_BATCH_INSPECT: 5,
+  CAMERA_LAYER_VIDEO_TEXTURE_TARGET: 6
+};
+
+export const LayerMasks = {
```
Comment (netpro2k) [inline src/components/layers.js:9]: Hmm. What do we need this for? The THREE.Layers interface has methods for flipping individual bits.

**Review by netpro2k — COMMENTED** [review]
I think we might be able to ditch LayerMasks, but other than that lgtm

File: src/systems/camera-system.js (hunk 2 above)
Comment (keianhzo) [reply src/systems/camera-system.js]: That's true. I'm going to switch to enable and remove the Layers mask enum.

File: src/components/video-texture-target.js (hunk 1 above)
Comment (keianhzo) [reply src/components/video-texture-target.js]: Updated

**Pull request merged**